[ODOO-414][ADD] Added new view description to control panel area

Jessica-BlueStingray opened the pull request:
### **[Add report description](https://metricwise.atlassian.net/browse/ODOO-414)**
This update adds a description area and `view_description` field to the Control Panel between the name and the Search Area.

To test:
Upgrade the modules
Turn on debug mode
Click on the debug button > Edit Action > Add a text to `view_description` > Save > Refresh page.

Jessica-BlueStingray commented:
@amh-mw Travis reminded me that I should had made the `view_description` feature responsive to different screen sizes. I just added a commit that hides it when the screen is smaller. See video:  


https://user-images.githubusercontent.com/87671101/188955443-cb577089-6ba8-476b-94cd-90e6245c1b79.mp4

Jessica-BlueStingray commented:
@amh-mw PR is down to 1 commit now

amh-mw commented:
This doesn't seem to want to patch with either -p1 or -p2. This is an unfortunate side effect of how Odoo changes the file hierarchy on disk as part of the Python packaging process. The "easy" fix would be to break this into two pull requests and patch one each way, but... 🤮 Let me think about it.

amh-mw reviewed (dismissed):
So after carefully reviewing the patch -p1 output, I think the best course of action is to take the server side changes to ir.actions and move them into a separate odoo-metricwise pull requests -- see my untested ODOO-414 branch for a head start.

amh-mw commented on odoo/addons/base/views/ir_actions_views.xml:
```diff
@@ -420,7 +421,7 @@ env['res.partner'].create({'name': partner_name})
                         <button name="action_launch"
                             states="open" string="Launch"
-                            type="object" icon="fa-cogs" class="oe_highlight" 
+                            type="object" icon="fa-cogs" class="oe_highlight"
```
Nit: No whitespace cleanup since we're never going to merge this upstream.

amh-mw commented:
Just as a heads up, I am working up a new patching modality that may solve this without any changes. Stay tuned.

Jessica-BlueStingray commented:
> Just as a heads up, I am working up a new patching modality that may solve this without any changes. Stay tuned.

Sorry I was really busy. 
I'm currently waiting for the news on the new patching then. If that doesn't solve it I'll do the other changes required.

amh-mw commented:
Ok, so the latest develop I just pushed adds a new `patch.sh` script that uses the GitHub CLI tool `gh` to figure out what patches to pull, then `sed` to modify their path hints so that -p1 works for all of them.

To test your patch, you can basically extract 

```
DIR="docker/odoo/patches"
NUM="20"
URL="https://patch-diff.githubusercontent.com/raw/MetricWise/odoo/pull"

curl --output-dir $DIR --remote-name --silent --url "$URL/$NUM.diff"
sed -E -I '' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"
```

And then `./run.sh -b` or `./test.sh -b` and if it looks good, `git add -u docker/odoo/patches/20.diff`

Jessica-BlueStingray commented:
@amh-mw The file was added correctly but I can't get past the `sed` command, I believe that's due to platform dependency, I'm using **Linux**. So I couldn't finish testing yet. Is there a `sed` command _linux version_ I can use?

amh-mw commented:
I can't say that I have ever used any unix flavor that didn't have [sed](https://linux.die.net/man/1/sed). It's been around [since 1973](https://en.wikipedia.org/wiki/Sed). 😁

Centos 6:
```
$ yum whatprovides `which sed`
sed-4.2.2-5.el7.x86_64 : A GNU stream text editor
```

Debian 11:
```
$ dpkg -l sed
sed            4.7-1        amd64        GNU stream editor for filtering/transforming text
```

amh-mw commented:
Ah... so the issue is that your flavor of sed doesn't have -E for ERE? I'll just have to rewrite the pattern as BRE.

Jessica-BlueStingray commented:
I'm not sure what the issue is, but the `sed` is not locating the file, although the file has been added correctly. 
_I think_ it is due to how the command is written, it might need to be different for linux. Sorry for my poor explanation earlier.
![Screenshot from 2022-09-16 14-59-48](https://user-images.githubusercontent.com/87671101/190722759-e9c00f53-8203-4018-a80e-b1b6fa2e356c.png)

amh-mw commented:
I've force pushed a BRE-compatible script to develop. Could you type `which sed`

Jessica-BlueStingray commented:
> I've force pushed a BRE-compatible script to develop. Could you type `which sed`

Yes, I get `/usr/bin/sed`

amh-mw commented:
Try the new patterns?
```
sed -i '' -e 's%^\(--- a\)/odoo%\1%' "$DIR/$NUM.diff"
sed -i '' -e 's%^\(+++ b\)/odoo%\1%' "$DIR/$NUM.diff"
```

Jessica-BlueStingray commented:
Still nothing

amh-mw commented:
What flavor of unix are you running? Maybe I can stand up a container to try and reproduce?

Jessica-BlueStingray commented:
Linux 5.15.0-46-generic
Ubuntu 20.04.4 LTS

amh-mw commented:
```
docker run ubuntu:focal sed --help

  -E, -r, --regexp-extended
                 use extended regular expressions in the script
                 (for portability use POSIX -E).
```

So you do have access to ERE. Maybe that's not the problem. Is your source code volume mounted read-only?

Jessica-BlueStingray commented:
> Is your source code volume mounted read-only?

No, it's not

Jessica-BlueStingray commented:
@amh-mw is there anything else I can to do to help here?

Jessica-BlueStingray commented:
@amh-mw  we managed it. It ran successfully.
The `sed` used in Ubuntu is `gsed (GNU)` for MAC, and command line used was slightly different than the original:

original: `sed -E -I '' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"`
used:    `sed -E -i'' -e 's%^(--- a|\+\+\+ b)/odoo%\1%' "$DIR/$NUM.diff"`

We removed the silent flag from the `curl` command line and found out that it was silently failing too due to `https` not being supported, then we changed to `http` and it was ok after that.

Both test and run were successful. Is there anything else we can do?

amh-mw commented:
You should create a pull request against odoo-metricwise, now that your patch has been copied into the docker/odoo/patches folder.

Jessica-BlueStingray commented:
> You should create a pull request against odoo-metricwise, now that your patch has been copied into the docker/odoo/patches folder.

I don't think I have permissions..

![Screenshot from 2022-09-22 17-00-55](https://user-images.githubusercontent.com/87671101/191859259-a1de4659-2f41-4e42-bcf5-ce0d2b684b82.png)

Jessica-BlueStingray commented:
Like this? : [Metricwise#11](https://github.com/MetricWise/odoo-metricwise/pull/11)

amh-mw commented:
Moved into #27.
